data-serving/data-service/src/model/revision-metadata.ts

```diff
@@ -11,20 +23,29 @@ export const revisionMetadataSchema = new mongoose.Schema({
         },
         required: 'Enter a revision id',
     },
-    moderator: {
-        type: String,
-        required: 'Enter a revision moderator id',
+    creationMetadata: {
+        type: editMetadataSchema,
+        required: 'Enter creation metadata',
     },
-    date: {
-        ...dateFieldInfo,
-        required: 'Enter a revision date',
+    updateMetadata: {
+        type: editMetadataSchema,
+        required: [
+            function (this: RevisionMetadataDocument): boolean {
+                return this.revisionNumber > 0 && !this.updateMetadata;
+            },
+            'Enter update metadata',
```

Comment:
I've wondered -- is it required to supply text to `required`? If not, does this type of comment provide value?

Reply:
good call, prob not, I'm actually going to remove it for all and just replace with required: true.